pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -659,6 +652,158 @@ contract ManagedPool is ManagedPoolSettings {
+        uint256 balanceTokenOut,
+        uint256 amountCalculated
+    ) private view {
+        uint256 actualSupply = _getActualSupply(_getVirtualSupply());
+
+        (uint256 amountIn, uint256 amountOut) = request.kind == IVault.SwapKind.GIVEN_IN
+            ? (request.amount, amountCalculated)
+            : (amountCalculated, request.amount);
+
+        // Since the balance of tokenIn is increasing, its BPT price will decrease,
+        // so we need to check the lower bound.
+        _checkCircuitBreaker(
+            BoundCheckKind.LOWER,
+            request.tokenIn,
+            actualSupply,
+            balanceTokenIn.add(amountIn),
+            tokenData.tokenInWeight
+        );
+
+        // Since the balance of tokenOut is decreasing, its BPT price will increase,
+        // so we need to check the upper bound.
+        _checkCircuitBreaker(
+            BoundCheckKind.UPPER,
+            request.tokenOut,
+            actualSupply,
+            balanceTokenOut.sub(amountOut),
+            tokenData.tokenOutWeight
+        );
+    }
+
+    /**
+     * @dev We need to check the breakers for all tokens on joins and exits (including join and exit swaps), since any
+     * change to the BPT supply affects all BPT prices. For a multi-token join or exit, we will have a set of
+     * balances and delta amounts. For a join/exitSwap, only one token balance is changing. We can use the same
+     * data for both: in the single token swap case, the other token `amounts` will be zero.
+     */
+    function _checkCircuitBreakersOnJoinOrExitSwap(
+        SwapRequest memory request,
+        uint256 actualSupply,
+        uint256 amountCalculated,
+        bool isJoin
+    ) private view {
+        uint256 newActualSupply;
+        uint256 amount;
+
+        // This is a swap between the BPT token and another pool token. Calculate the end state: actualSupply
+        // and the token amount being swapped, depending on whether it is a join or exit, GivenIn or GivenOut.
+        if (isJoin) {
+            (newActualSupply, amount) = request.kind == IVault.SwapKind.GIVEN_IN
+                ? (actualSupply.add(amountCalculated), request.amount)
+                : (actualSupply.add(request.amount), amountCalculated);
+        } else {
+            (newActualSupply, amount) = request.kind == IVault.SwapKind.GIVEN_IN
+                ? (actualSupply.sub(request.amount), amountCalculated)
+                : (actualSupply.sub(amountCalculated), request.amount);
+        }
+
+        // Since this is a swap, we do not have all the tokens, balances, or weights, and need to fetch them.
+        (IERC20[] memory tokens, uint256[] memory balances) = _getPoolTokens();
+        uint256[] memory normalizedWeights = _getNormalizedWeights(tokens);
+        uint256[] memory scalingFactors = _scalingFactors(tokens);
+        _upscaleArray(balances, scalingFactors);
+
+        // Initialize to all zeros, and set the deltaAmount associated with the swap.
```

Comment:
We don't make use of the term `deltaAmount` anywhere else. We should update this comment.